jupyterlab_latex/__init__.py

```diff
@@ -43,45 +44,74 @@ class LatexConfig(Configurable):
     """
     latex_command = Unicode('xelatex', config=True,
         help='The LaTeX command to use when compiling ".tex" files.')
+    bib_command = Unicode('bibtex', config=True,
+        help='The bibTeX command to use when compiling ".tex" files.')
 
 
 class LatexHandler(APIHandler):
     """
     A handler that runs LaTeX on the server.
     """
+    
+    
+    @gen.coroutine
+    def run_latex(self, tex_dir, tex_base_name, whitelist):
+        c = LatexConfig(config=self.config)
+        
+        with latex_cleanup(
+            workdir=tex_dir,
+            whitelist=whitelist
+            ):
+            full_latex_sequence = [
+                c.latex_command,
+                "-interaction=nonstopmode",
+                "-halt-on-error",
+                "-file-line-error",
+                f"{tex_base_name}",
+                ]
+            full_bibtex_sequence = [
+                c.bib_command,
+                f"{tex_base_name}",
+                ]
+            command_sequence = [list(full_latex_sequence)]
+            
+            if any([re.match(r'.*\.bib', x) for x in set(glob.glob("*"))]):
+                command_sequence += [
+                    list(full_bibtex_sequence), 
+                    list(full_latex_sequence), 
+                    list(full_latex_sequence),
+                    ]
+                    
+            for i, cmd in enumerate(command_sequence):
```

Comment:
It does not look like you are using `i` here, so we might as well do `for cmd in command`

Reply:
I was actually using the i inside the error but I can get around that.

Reply:
Ah, my mistake! Consider this comment null and void, then.